### PR TITLE
add voice ai classes that access elevenlabs ap

### LINKED ISSUE #33: voice support for eleven labs

### PR DESCRIPTION
**Pull request by ephraimacquah**
closes #33

### CODE REVIEW

**Comment by ephraimacquah** [comment]
test_ai_presenter.py needs updated to pass in a voice ai into that instance of aipresenter to pass tests

File: ai_presenter/voice_ai/elevenlabs.py [hunk 1]
```diff
@@ -1,24 +1,69 @@
...
+    # actor.says returns audio output
+    # this is concatenated with previous actor.says outputs
+    # output file opened and audio output is written to output_file
     # this is saved into output file
     # return output file
 
-    def generate(self, input_file, output_file, c: VoiceConfig):
+    def generate(self, input_file: str, output_file: str):
         logging.info('Generating audio file')
+        with open(input_file) as file:
```
Comment (lpabon) [inline ai_presenter/voice_ai/elevenlabs.py:55]: Use readlines to read each line of the input file as a json string

File: ai_presenter/voice_ai/fake.py [hunk 2]
```diff
@@ -37,18 +50,30 @@ def new_actor(self, config):
...
         logging.info(f'VoiceAIFake: Opening input file: {input_file} and ' +
                      'extracting send info')
         with open(input_file, 'r') as input:
-            text = input.read()
+            data = json.load(input)
 
         logging.info('VoiceAIFake: Generating audio file')
+        audio = ''
+        for message in data['dialogue']:
+            character_config = VoiceConfig(message['speaker'],
```
Comment (lpabon) [inline ai_presenter/voice_ai/fake.py:64]: Check a `self.characters = {}` to see if created already. If not, then:

```
name = message['speaker']
self.characters[name] = self.new_actor(VoiceConfig(...))
....
self.characters[name].says(...)
```

**Review by lpabon — CHANGES_REQUESTED** [review]

File: ai_presenter/voice_ai/base.py [hunk 3]
```diff
@@ -1,28 +1,56 @@
...
+    def create_character_db(self, input_file: str):
+        with open(input_file, 'r') as input:
+            for line in input:
+                json_string = line.strip()
+                data = (json.loads(json_string))
+
+        for message in data['dialogue']:
+            name = message['speaker']
+            if name not in self.characters:
+                character_config = VoiceConfig(name,
```
Comment (ephraimacquah) [inline ai_presenter/voice_ai/base.py:49]: kept it in this format to keep from going over 80 characters per line

File: ai_presenter/voice_ai/elevenlabs.py [hunk 4]
```diff
@@ -1,24 +1,69 @@
...
+            data = json.load(file)
```
[the remark this answers is not in the document]
Comment (lpabon) [reply ai_presenter/voice_ai/elevenlabs.py]: Interesting. Let's take a look at the input file on Monday

[the remark this answers is not in the document]
Comment (ephraimacquah) [reply ai_presenter/voice_ai/elevenlabs.py]: The readlines method of implementation to parse the file fixed this issue

File: ai_presenter/voice_ai/base.py [hunk 5]
```diff
@@ -1,28 +1,56 @@
...
     def new_actor(self, config):
         logging.info("New actor created")
+
+    def create_character_db(self, input_file: str):
+        with open(input_file, 'r') as input:
+            for line in input:
+                json_string = line.strip()
+                data = (json.loads(json_string))
+
+        for message in data['dialogue']:
```
Comment (lpabon) [inline ai_presenter/voice_ai/base.py]: Check the tab here so that this is in the for loop. It should be under data.

**Review by lpabon — APPROVED** [review]

**Review by lpabon — APPROVED** [review]

**Pull request merged**